diff --git a/arch/arm/armfeature.c b/arch/arm/armfeature.c
index 326bd07..0c36a81 100644
--- a/arch/arm/armfeature.c
+++ b/arch/arm/armfeature.c
@@ -37,8 +37,8 @@ static inline int arm_has_neon()
 }
 #endif
 
-ZLIB_INTERNAL int arm_cpu_has_neon;
-ZLIB_INTERNAL int arm_cpu_has_crc32;
+ZLIB_INTERNAL int arm_cpu_has_neon = 1;
+ZLIB_INTERNAL int arm_cpu_has_crc32 = 0;
 
 static void __attribute__((constructor)) arm_check_features(void) {
 #if defined(__aarch64__) || defined(_M_ARM64)
