diff --git a/Foundation/src/File_UNIX.cpp b/Foundation/src/File_UNIX.cpp
index 7878f0294..1fcbe1750 100644
--- a/Foundation/src/File_UNIX.cpp
+++ b/Foundation/src/File_UNIX.cpp
@@ -213,8 +213,8 @@ Timestamp FileImpl::createdImpl() const
 	poco_assert (!_path.empty());
 
 #if defined(__APPLE__) && defined(st_birthtime) && !defined(POCO_NO_STAT64) // st_birthtime is available only on 10.5
-	struct stat64 st;
-	if (stat64(_path.c_str(), &st) == 0)
+	struct stat st;
+	if (stat(_path.c_str(), &st) == 0)
 		return Timestamp::fromEpochTime(st.st_birthtime);
 #elif defined(__FreeBSD__)
 	struct stat st;
