diff --git a/libs/context/src/asm/jump_arm64_aapcs_elf_gas.S b/libs/context/src/asm/jump_arm64_aapcs_elf_gas.S
index cefd183..d3ee2c2 100644
--- a/libs/context/src/asm/jump_arm64_aapcs_elf_gas.S
+++ b/libs/context/src/asm/jump_arm64_aapcs_elf_gas.S
@@ -55,7 +55,7 @@
 .text
 .align  2
 .global jump_fcontext
-.type   jump_fcontext, %function
+// .type   jump_fcontext, %function
 jump_fcontext:
     # prepare stack for GP + FPU
     sub  sp, sp, #0xb0
@@ -109,6 +109,6 @@ jump_fcontext:
     add  sp, sp, #0xb0
 
     ret x4
-.size   jump_fcontext,.-jump_fcontext
+// .size   jump_fcontext,.-jump_fcontext
 # Mark that we don't need executable stack.
-.section .note.GNU-stack,"",%progbits
+// .section .note.GNU-stack,"",%progbits
diff --git a/libs/context/src/asm/make_arm64_aapcs_elf_gas.S b/libs/context/src/asm/make_arm64_aapcs_elf_gas.S
index 66cfb2d..bf891af 100644
--- a/libs/context/src/asm/make_arm64_aapcs_elf_gas.S
+++ b/libs/context/src/asm/make_arm64_aapcs_elf_gas.S
@@ -55,7 +55,7 @@
 .text
 .align  2
 .global make_fcontext
-.type   make_fcontext, %function
+/* .type   make_fcontext, %function */
 make_fcontext:
     # shift address in x0 (allocated stack) to lower 16 byte boundary
     and x0, x0, ~0xF
@@ -80,6 +80,6 @@ finish:
     # exit application
     bl  _exit
 
-.size   make_fcontext,.-make_fcontext
+// .size   make_fcontext,.-make_fcontext
 # Mark that we don't need executable stack.
-.section .note.GNU-stack,"",%progbits
+// .section .note.GNU-stack,"",%progbits
diff --git a/libs/context/src/asm/ontop_arm64_aapcs_elf_gas.S b/libs/context/src/asm/ontop_arm64_aapcs_elf_gas.S
index 665ca5a..b1efa8e 100644
--- a/libs/context/src/asm/ontop_arm64_aapcs_elf_gas.S
+++ b/libs/context/src/asm/ontop_arm64_aapcs_elf_gas.S
@@ -55,7 +55,7 @@
 .text
 .align  2
 .global ontop_fcontext
-.type   ontop_fcontext, %function
+// .type   ontop_fcontext, %function
 ontop_fcontext:
     # prepare stack for GP + FPU
     sub  sp, sp, #0xb0
@@ -108,6 +108,6 @@ ontop_fcontext:
 
     # jump to ontop-function
     ret x2
-.size   ontop_fcontext,.-ontop_fcontext
+// .size   ontop_fcontext,.-ontop_fcontext
 # Mark that we don't need executable stack.
-.section .note.GNU-stack,"",%progbits
+// .section .note.GNU-stack,"",%progbits
